clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -10804,12 +10799,21 @@ void LinkerWrapper::ConstructJob(Compilation &C, const JobAction &JA,
     for (auto &I :
          llvm::make_range(ToolChainRange.first, ToolChainRange.second)) {
       const ToolChain *TC = I.second;
-      if (TC->getTriple().isSPIR() &&
+      if (TC->getTriple().isSPIROrSPIRV() &&
           TC->getTriple().getSubArch() == llvm::Triple::NoSubArch) {
         TargetTriple = TC->getTriple();
         break;
       }
     }
+    // Pass the device triple to the linker wrapper tool for SYCL offload.
+    // Only spir64 or spirv64 is currently passed.
+    // TODO(NOM1): Support target triples in a more generic way.
+    // TODO(NOM3): Investigate why passing spirv64-unknown-unknown does not
+    // work.
+    if (TargetTriple.isSPIR())
+      CmdArgs.push_back("--triple=spir64");
+    else
```

Comment:
I know this is inside a sycl if, but could we an an assert in the else case to confirm it's SPIR-V and not some other arch like NVPTX or something. Not sure if that can happen but the else worries me.

Reply:
Good catch @sarnex Thanks. I added a check for spirv.